### DISCUSSION TITLE
Font paths broken in production

### DISCUSSION THREAD

**Discussion by jjla26** [shakacode/shakapacker#362] [General]
I will create this discussion here because I don't know if this is an intended behaviour or a bug: 

From the latest update to shakapacker 7 we have a problem with some assets in production. We are using a package for math formulas https://www.npmjs.com/package/katex and after the update the fonts of the formulas were broken. Then we realised that the  fonts were being loaded from the "wrong" path

`https://cloudfront.net/shared/node_modules/katex/dist/fonts/KaTeX_Main-Regular-13b3f8528a571d1cb797eb520002dde9.woff2`

instead of 

`https://cloudfront.net/packs/shared/node_modules/katex/dist/fonts/KaTeX_Main-Regular-13b3f8528a571d1cb797eb520002dde9.woff2`

which was causing some CORS issues. The fonts built in a different path was caused by this commit https://github.com/shakacode/shakapacker/pull/283/files#diff-716197750a05276089fff034fe03597e5d569526a8ea683fb0022cf69456cbc4 that we had to revert in our config to fix the fonts. Is this assets path the way these fonts should be loaded now?

## Expected behavior:
Fonts build within the same path in production `https://cloudfront.net/packs/shared/node_modules/katex/dist/fonts/KaTeX_Main-Regular-13b3f8528a571d1cb797eb520002dde9.woff2`

## Actual behavior:
Fonts build in a different path `https://cloudfront.net/shared/node_modules/katex/dist/fonts/KaTeX_Main-Regular-13b3f8528a571d1cb797eb520002dde9.woff2`

**Comment by ahangarha** [comment]
This change shouldn't make any difference between different environments. Could you please recheck the behavior? (of course, by removing the bundled assets in the dev environment)

**Reply by jjla26** [reply]
You are right. There is no difference between environments. But that commit changes the path where the font assets are,  which is causing some CORS problems in production (I know the CORS problems we could solved changing the policy). I could be a little bit more specific in my quesiton:

About this commit https://github.com/shakacode/shakapacker/pull/283/files#diff-716197750a05276089fff034fe03597e5d569526a8ea683fb0022cf69456cbc4 -> I can see that the `.slice(1)` was changed to a `.filter(Boolean)` which are doing 2 different things. Before the filename was something was in 'node_modules/katex/dist/fonts/KaTeX_Caligraphic-Bold.woff2' and after the processing and the `.slice(1)` we were getting this: [ 'katex', 'dist', 'fonts' ]

Now without the slice we get [ 'node_modules', 'katex', 'dist', 'fonts' ] which changes the path.

So my question is: is this change intended to be like this ? to change the paths where those assets are ?

**Reply by Judahmeek** [reply]
@jjla26 the removal of `slice(1)` was an intentional breaking change.

However, I don't think that it explains the issue that you're having.

The removal of `slice(1)` would possibly result in a directory being added to your path.

The issue you have is that the `packs` directory is missing from your path.
